File: frontend/src/app/decision/services/decision.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Decision } from '../models/decision.model';
import { environment } from '../../../environments/environment';

@Injectable({
  providedIn: 'root'
})
export class DecisionService {
  private apiUrl = `${environment.apiUrl}/decisions`;

  constructor(private http: HttpClient) {}

  createDecision(decision: Partial<Decision>): Observable<Decision> {
    return this.http.post<Decision>(this.apiUrl, decision);
  }

  updateDecision(decisionId: number, decision: Partial<Decision>): Observable<Decision> {
    return this.http.put<Decision>(`${this.apiUrl}/${decisionId}`, decision);
  }

  deleteDecision(decisionId: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${decisionId}`);
  }

  getDecisionById(decisionId: number): Observable<Decision> {
    return this.http.get<Decision>(`${this.apiUrl}/${decisionId}`);
  }

  getAllDecisions(): Observable<Decision[]> {
    return this.http.get<Decision[]>(this.apiUrl);
  }

  getDecisionsByMeeting(meetingId: number): Observable<Decision[]> {
    return this.http.get<Decision[]>(`${this.apiUrl}/meeting/${meetingId}`);
  }

  getDecisionsByProposer(userId: number): Observable<Decision[]> {
    return this.http.get<Decision[]>(`${this.apiUrl}/proposer/${userId}`);
  }

  getPendingDecisionsByMeeting(meetingId: number): Observable<Decision[]> {
    return this.http.get<Decision[]>(`${this.apiUrl}/meeting/${meetingId}/pending`);
  }

  updateDecisionStatus(decisionId: number, status: string): Observable<Decision> {
    return this.http.patch<Decision>(`${this.apiUrl}/${decisionId}/status`, { status });
  }
} 

File: backend/src/main/java/com/tunisair/meetingmanagement/service/DecisionService.java
package com.tunisair.meetingmanagement.service;

import com.tunisair.meetingmanagement.model.Decision;
import com.tunisair.meetingmanagement.model.Meeting;
import com.tunisair.meetingmanagement.model.User;

import java.time.LocalDateTime;
import java.util.List;
import java.util.Optional;

public interface DecisionService {
    List<Decision> getAllDecisions();
    Optional<Decision> getDecisionById(Long id);
    Decision createDecision(Decision decision);
    Decision updateDecision(Long id, Decision decisionDetails);
    void deleteDecision(Long id);
    List<Decision> getDecisionsByMeeting(Meeting meeting);
    List<Decision> getDecisionsByResponsibleUser(User user);
    List<Decision> getDecisionsByDeadlineRange(LocalDateTime start, LocalDateTime end);
    List<Decision> getPendingDecisions();
    Decision assignResponsibleUser(Long decisionId, Long userId);
    Decision updateDeadline(Long decisionId, LocalDateTime newDeadline);
} 